packages/components/pie-modal/src/defs.ts

```diff
@@ -1,8 +1,19 @@
 export const headingLevels = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'] as const;
 
 export interface ModalProps {
+    /**
+     * The text to display in the modal's heading.
```

Comment:
Is it worth noting which of these are required and optional? (I think there's a `@required` directive if so).

If not required, I'd explicitly say here that the heading is optional.

Reply:
Conflicted about this. Seems super smelly to not leverage actual type systems to achieve this. Not sure how we go about it though. Can we look into that separately? I can't see a `@required` directive available

Reply:
Sounds good, but do you mind pointing me to an example of this in the wild? I honestly can't seem to see how we could achieve this!